artwork/Modules/Budget/Services/BudgetService.php

```diff
@@ -6,116 +6,119 @@
-            table: $table,
-            budgetTypesEnum: BudgetTypesEnum::BUDGET_TYPE_EARNING,
-            name: 'Hauptpostion',
-            position: $table->mainPositions()->where('type', BudgetTypesEnum::BUDGET_TYPE_EARNING)->max('position') + 1
-        );
-
-
-
-        $costSubPosition = $costMainPosition->subPositions()->create([
-            'name' => 'Unterposition',
-            'position' => $costMainPosition->subPositions()->max('position') + 1
-        ]);
-
-        $earningSubPosition = $earningMainPosition->subPositions()->create([
-            'name' => 'Unterposition',
-            'position' => $earningMainPosition->subPositions()->max('position') + 1
-        ]);
-
-        $costSubPositionRow = $costSubPosition->subPositionRows()->create([
-            'commented' => false,
-            'position' => $costSubPosition->subPositionRows()->max('position') + 1
-        ]);
-
-        $earningSubPositionRow = $earningSubPosition->subPositionRows()->create([
-            'commented' => false,
-            'position' => $earningSubPosition->subPositionRows()->max('position') + 1
-
-        ]);
-
-        $firstThreeColumns = $columns->shift(3);
-
-        $costSubPositionRow->columns()->attach($firstThreeColumns->pluck('id'), [
-            'value' => "",
-            'verified_value' => "",
-            'linked_money_source_id' => null,
-        ]);
-
-        $earningSubPositionRow->columns()->attach($firstThreeColumns->pluck('id'), [
-            'value' => "",
-            'verified_value' => "",
-            'linked_money_source_id' => null,
-        ]);
-
-        $costSubPositionRow->columns()->attach($columns->pluck('id'), [
-            'value' => 0,
-            'verified_value' => null,
-            'linked_money_source_id' => null,
-        ]);
-
-        $costMainPosition->mainPositionSumDetails()->create([
-            'column_id' => $columns->first()->id
-        ]);
-
-        $earningMainPosition->mainPositionSumDetails()->create([
-            'column_id' => $columns->first()->id
-        ]);
-
-        $costSubPosition->subPositionSumDetails()->create([
-            'column_id' => $columns->first()->id
-        ]);
-
-        $earningSubPosition->subPositionSumDetails()->create([
-            'column_id' => $columns->first()->id
-        ]);
-
-        BudgetSumDetails::create([
-            'column_id' => $columns->first()->id,
-            'type' => 'COST'
-        ]);
-
-        BudgetSumDetails::create([
-            'column_id' => $columns->first()->id,
-            'type' => 'EARNING'
-        ]);
+    }
 
-        $earningSubPositionRow->columns()->attach($columns->pluck('id'), [
-            'value' => 0,
-            'verified_value' => null,
-            'linked_money_source_id' => null,
-        ]);
 
+    public function generateBasicBudgetValues(Project $project): void
+    {
+        DB::transaction(function() use ($project) {
```

Comment:
If you have multiple things happening in the database its best to use a transaction to prevent data garbe is one statement fails